### ISSUE TITLE
Crashes when has to do a lot of change at once deeply nested.

### ISSUE THREAD

**Issue by sbimochan** [JannisX11/batch-rename#3]

**Comment by Edduard** [comment]
I can confirm this. 

Also happened to me. I was trying to rename about 130 folders/subfolders/files and after saving the file with names it just replaced the first folder. than doing it again would replace just the next folder in line and so on. doesn't throw any error.

I'm using Windows 11 if it helps.

**Comment by teneko** [comment]
Yes, I have noticed this too, @Edduard. That’s why I started implementing a more [hierarchical-friendly "batch rename" vscode extension](https://marketplace.visualstudio.com/items?itemName=teneko.hierarchical-batch-rename). It supports more complex rename operations, though it still doesn’t handle merges or cyclic moves. But at least you can now rename files and folders across multiple different hierarchies in one go. 🙂
